google-cloud-storage/src/main/java/com/google/cloud/storage/BlobWriteChannel.java

```diff
@@ -17,47 +17,41 @@
 package com.google.cloud.storage;
 
 import static com.google.cloud.RetryHelper.runWithRetries;
+import static java.util.Objects.requireNonNull;
 import static java.util.concurrent.Executors.callable;
 
 import com.google.api.services.storage.model.StorageObject;
 import com.google.cloud.BaseWriteChannel;
+import com.google.cloud.ExceptionHandler;
 import com.google.cloud.RestorableState;
 import com.google.cloud.RetryHelper;
 import com.google.cloud.WriteChannel;
-import com.google.cloud.storage.spi.v1.StorageRpc;
 import java.math.BigInteger;
-import java.net.URL;
-import java.util.Map;
-import java.util.concurrent.Callable;
+import java.util.function.Supplier;
+import org.checkerframework.checker.nullness.qual.NonNull;
 
 /** Write channel implementation to upload Google Cloud Storage blobs. */
 class BlobWriteChannel extends BaseWriteChannel<StorageOptions, BlobInfo> {
 
-  BlobWriteChannel(StorageOptions options, BlobInfo blob, Map<StorageRpc.Option, ?> optionsMap) {
```

Comment:
How are we able to remove these methods?

Reply:
Because there are differing retry handling needs for open vs write on a resumable session and because BlobWriteChannel implements Restorable, I separated open out into ResumableMedia helper class which allowed for the removal of these overloaded constructors.

Reply:
Gotcha, just wanted to make sure that this isn't something affecting the public surface of the library!

Reply:
No worries, that's a totally valid concern. These constructors (and even this whole class) is package private so isn't part of the public api today.